vignettes/adam-adadas.Rmd

```diff
@@ -0,0 +1,155 @@
+---
+title: "CDISC Pilot ADaM: ADAS Cog Analysis (ADADAS)"
+output:
+  html_document:
+    df_print: paged
+    toc: yes
+    toc_depth: '2'
+    toc_float: true
+---
+
+```{r, message = FALSE}
+# Initiate start-up file
+source(file.path(rprojroot::find_root("DESCRIPTION"), "inst/startup.R"))
+```
+```{r}
+# Working directory requires write permission
+if (file.access(".", 2) != 0) {
+  warning(
+    "The working directory '", normalizePath("."), "' is not writable.\n",
+    "Please change it to a location with write permission."
+  )
+}
+```
+
+```{r setup, message=FALSE}
+# CRAN package, please using install.packages() to install
+library(dplyr)
+library(admiral)
+library(metacore)
+library(metatools)
+library(stringr)
+```
+
+
+## Step 1: Read in data (SDTM.QS, ADaM.ADSL)
+
+```{r}
+qs <- haven::read_xpt(file.path(path$sdtm, "qs.xpt"))
+adsl <- haven::read_xpt(file.path(path$adam, "adsl.xpt"))
+adas_qc <- haven::read_xpt(file.path(path$adam, "adadas.xpt"))
+```
+
+## Step 2: Convert blanks to NA
+```{r}
+qs <- convert_blanks_to_na(qs)
+```
+
+## Step 3:
+```{r}
+## placeholder for origin=predecessor, use metatool::build_from_derived()
+
+## derive ADT/ADY
+# Get list of ADSL vars required for derivations
+adsl_vars <- vars(TRTSDT, TRTEDT, TRT01A, TRT01P)
+adas1 <- qs %>%
+  # subset to interested QSTESTCD
+  filter(QSTESTCD %in%
+    c(str_c("ACITM", str_pad(1:14, 2, pad = "0")), "ACTOT")) %>%
+  # Join ADSL with QS (need TRTSDT for ADY derivation)
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by_vars = vars(STUDYID, USUBJID)
+  ) %>%
+  ## Calculate ADT, ADY ----
+  derive_vars_dt(
+    new_vars_prefix = "A",
+    dtc = QSDTC
+  ) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = vars(ADT))
+
+## derive AVISIT/AVISITN
+adas2 <- adas1 %>%
+  mutate(
+    AVISIT = case_when(
+      ADY <= 1 ~ "Baseline",
+      ADY >= 2 & ADY <= 84 ~ "Week 8",
+      ADY >= 85 & ADY <= 140 ~ "Week 16",
+      ADY > 140 ~ "Week 24",
+      TRUE ~ NA_character_
+    ),
+    PARAMCD = QSTESTCD
+  )
+```
+
+## PARAM, PARAMN
+```{r}
+## placeholder: replace derive_vars_merged_lookup by metatool::create_var_from_codelist()/create_cat_var()
+## Add PARAMCD PARAM and PARAMN - from LOOK-UP table ----
+# Replace derive_vars_merged_lookup() with PARAMCD lookup function
+```
+ 
+## AWRANGE/AWTARGET/AWTDIFF/AWLO/AWHI/AWU
+```{r}
+## derive AWRANGE/AWTARGET/AWTDIFF/AWLO/AWHI/AWU
+aw_lookup <- tribble(
+  ~AVISIT, ~AWRANGE, ~AWTARGET, ~AWLO, ~AWHI,
+  "Baseline", "<=1", 1, NA_integer_, 1,
+  "Week 8", "2-84", 56, 2, 84,
+  "Week 16", "85-140", 112, 85, 140,
+  "Week 24", ">140", 168, 141, NA_integer_
+)
+
+adas3 <- derive_vars_merged(
+  adas2,
+  dataset_add = aw_lookup,
+  by_vars = vars(AVISIT)
+) %>%
+  mutate(
+    AWTDIFF = abs(AWTARGET - ADY),
+    AWU = "DAYS"
+  )
```

Comment:
A bit overkill but please use `derive_vars_duration()`.

Reply:
`derive_vars_duration()` is not applicable here:
* `derive_vars_duration()`  derives duration from **dates**, but AWTARGET is the target day within the visit window defined in SAP, not dates;
* AWTDIFF is the absolute difference between AWTARGET - ADY, not duration.

Reply:
I see, never mind then 🙂